openedx/core/lib/xblock_builtin/xblock_discussion/xblock_discussion.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+"""
+Discussion XBlock
+"""
+import json
+import logging
+
+from xblockutils.resources import ResourceLoader
+from xblockutils.studio_editable import StudioEditableXBlockMixin
+
+from xblock.core import XBlock
+from xblock.fields import Scope, String, UNIQUE_ID
+from xblock.fragment import Fragment
+
+log = logging.getLogger(__name__)
+loader = ResourceLoader(__name__)  # pylint: disable=invalid-name
+
+
+def _(text):
+    """
+    A noop underscore function that marks strings for extraction.
+    """
+    return text
+
+
+@XBlock.needs('user')
+@XBlock.needs('i18n')
+class DiscussionXBlock(XBlock, StudioEditableXBlockMixin):
+    """
+    Provides a discussion forum that is inline with other content in the courseware.
+    """
+    discussion_id = String(scope=Scope.settings, default=UNIQUE_ID, force_export=True)
+    display_name = String(
+        display_name=_("Display Name"),
+        help=_("Display name for this module"),
+        default="Discussion",
+        scope=Scope.settings
+    )
+    discussion_category = String(
+        display_name=_("Category"),
+        default=_("Week 1"),
+        help=_(
+            "A category name for the discussion. "
+            "This name appears in the left pane of the discussion forum for the course."
+        ),
+        scope=Scope.settings
+    )
+    discussion_target = String(
+        display_name=_("Subcategory"),
+        default="Topic-Level Student-Visible Label",
+        help=_(
+            "A subcategory name for the discussion. "
+            "This name appears in the left pane of the discussion forum for the course."
+        ),
+        scope=Scope.settings
+    )
+    sort_key = String(scope=Scope.settings)
+
+    editable_fields = ["display_name", "discussion_category", "discussion_target"]
+
+    has_author_view = True  # Tells Studio to use author_view
+
+    @property
+    def course_key(self):
+        """
+        :return: int course id
+        """
+        return getattr(self.scope_ids.usage_id, 'course_key', None)
```

Comment:
@ormsbee it shouldn't be the case with this particular XBlock, but usually we use workbench (aka xblock-sdk) to run UI/integration tests on XBlocks, and it differs from LMS/Studio runtime quite a lot. I can't tell off top of my head if `runtime.course_id` is available in workbench (most likely not), but this code is here to compensate for those differences.

As I said, it's unlikely to be the issue here, but our team just got into habit of (almost) always writing "cross-runtime" code.

@pomegranited I would suggest trying the approach @ormsbee suggested, but adding a comment about possible issues in workbench runtime - it would be quite useful if someone ever moves this code out of edx-platform and adds workbench-based test.